dask_cuda/worker_spec.py

```diff
@@ -0,0 +1,128 @@
+import os
+
+from dask.distributed import Nanny
+from distributed.system import MEMORY_LIMIT
+
+from .local_cuda_cluster import cuda_visible_devices
+from .utils import CPUAffinity, get_cpu_affinity, get_gpu_count
+
+
+def worker_spec(
+    interface=None,
+    protocol=None,
+    dashboard_address=":8787",
+    threads_per_worker=1,
+    silence_logs=True,
+    CUDA_VISIBLE_DEVICES=None,
+    enable_infiniband=False,
+    ucx_net_devices="",
+    enable_nvlink=False,
+    **kwargs
+):
+    """ Create a Spec for a CUDA worker.
+
+    The Spec created by this function can be used as a recipe for CUDA workers
+    that can be passed to a SpecCluster.
+
+    Parameters
+    ----------
+    interface: str
+        The external interface used to connect to the scheduler.
+    protocol: str
+        The protocol to used for data transfer, e.g., "tcp" or "ucx".
+    dashboard_address: str
+        The address for the scheduler dashboard.  Defaults to ":8787".
+    threads_per_worker: int
+        Number of threads to be used for each CUDA worker process.
+    silence_logs: bool
+        Disable logging for all worker processes
+    CUDA_VISIBLE_DEVICES: str
+        String like ``"0,1,2,3"`` or ``[0, 1, 2, 3]`` to restrict activity to
+        different GPUs
+    enable_infiniband: bool
+        Set environment variables to enable UCX InfiniBand support
+    ucx_net_device: str or callable
+        A string with the interface name to be used for all devices (empty
+        string means use default), or a callable function taking an integer
+        identifying the GPU index.
```

Comment:
This seems very specific.   I wonder if there is some more general solution, like a function that produces an environment or something that might be more flexible to future use cases.

Reply:
We need to find something more generic, but I don't really have the time to do it now. So I think this is an ok-ish solution until we can come up with something that works better.